test: `acceptstalefeeestimates` failure in `feature_fee_estimation` after duplicate coinbase tx weight reservation fix

### Summary
First encountered in https://github.com/bitcoin/bitcoin/actions/runs/14929268797/job/41941137828?pr=31144

Doing a git bisect reveals the likely culprit is [mining: bugfix: Fix duplicate coinbase tx weight reservation](https://github.com/bitcoin/bitcoin/pull/31384)

---

### Reproducer

> git checkout 6b165f5906fc53bd10bedff85a6ef26e0aabdc5c

```bash
HEAD is now at 6b165f5906 Merge bitcoin/bitcoin#31384: mining: bugfix: Fix duplicate coinbase tx weight reservation
```

> cmake -B build -DCMAKE_BUILD_TYPE=Release && cmake --build build -j$(nproc) && build/test/functional/feature_fee_estimation.py --random=3450808900320758527

```bash
2025-05-09T14:55:06.412000Z TestFramework (INFO): User supplied random seed 3450808900320758527
2025-05-09T14:55:06.412000Z TestFramework (INFO): PRNG seed is: 3450808900320758527
2025-05-09T14:55:06.413000Z TestFramework (INFO): Initializing test directory /var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_wg3p8s0v
2025-05-09T14:55:06.426000Z TestFramework (INFO): This test is time consuming, please be patient
2025-05-09T14:55:06.426000Z TestFramework (INFO): Splitting inputs so we can generate tx's
2025-05-09T14:55:06.988000Z TestFramework (INFO): Finished splitting
2025-05-09T14:55:07.791000Z TestFramework (INFO): Testing estimates with single transactions.
2025-05-09T14:55:07.791000Z TestFramework (INFO): Will output estimates for 1/2/3/6/15/25 blocks
2025-05-09T14:55:07.791000Z TestFramework (INFO): Creating transactions and mining them with a block size that can't keep up
2025-05-09T14:55:09.501000Z TestFramework (INFO): Creating transactions and mining them at a block size that is just big enough
2025-05-09T14:55:11.955000Z TestFramework (INFO): Creating transactions and mining them with a block size that can't keep up
2025-05-09T14:55:14.407000Z TestFramework (INFO): Creating transactions and mining them at a block size that is just big enough
2025-05-09T14:55:17.271000Z TestFramework (INFO): Final estimates after emptying mempools
2025-05-09T14:55:17.287000Z TestFramework (INFO): Test fee_estimates.dat is flushed periodically
2025-05-09T14:55:18.131000Z TestFramework (INFO): Test fee rate estimation after restarting node with high MempoolMinFee
2025-05-09T14:55:18.523000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/lorinc/IdeaProjects/bitcoin/test/functional/test_framework/test_framework.py", line 135, in main
    self.run_test()
    ~~~~~~~~~~~~~^^
  File "/Users/lorinc/IdeaProjects/bitcoin/build/test/functional/feature_fee_estimation.py", line 458, in run_test
    self.test_feerate_mempoolminfee()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/lorinc/IdeaProjects/bitcoin/build/test/functional/feature_fee_estimation.py", line 243, in test_feerate_mempoolminfee
    check_estimates(self.nodes[1], self.fees_per_kb)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lorinc/IdeaProjects/bitcoin/build/test/functional/feature_fee_estimation.py", line 121, in check_estimates
    check_smart_estimates(node, fees_seen)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/Users/lorinc/IdeaProjects/bitcoin/build/test/functional/feature_fee_estimation.py", line 104, in check_smart_estimates
    raise AssertionError(
        f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})"
    )
AssertionError: Estimated fee (0.00923427) out of range (5.076142131979696e-05,0.008705986394557823)
2025-05-09T14:55:18.578000Z TestFramework (INFO): Not stopping nodes as test failed. The dangling processes will be cleaned up later.
2025-05-09T14:55:18.578000Z TestFramework (WARNING): Not cleaning up dir /var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_wg3p8s0v
2025-05-09T14:55:18.578000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_wg3p8s0v/test_framework.log
2025-05-09T14:55:18.579000Z TestFramework (ERROR):
2025-05-09T14:55:18.579000Z TestFramework (ERROR): Hint: Call /Users/lorinc/IdeaProjects/bitcoin/test/functional/combine_logs.py '/var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_wg3p8s0v' to consolidate all logs
2025-05-09T14:55:18.579000Z TestFramework (ERROR):
2025-05-09T14:55:18.579000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2025-05-09T14:55:18.579000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2025-05-09T14:55:18.579000Z TestFramework (ERROR):
[node 2] Cleaning up leftover process
[node 1] Cleaning up leftover process
[node 0] Cleaning up leftover process
```

> git checkout HEAD^1

```bash
HEAD is now at 6a46be75c4 Merge bitcoin/bitcoin#31793: ci: Use clang-20 for sanitizer tasks
```

>  cmake -B build -DCMAKE_BUILD_TYPE=Release && cmake --build build -j$(nproc) && build/test/functional/feature_fee_estimation.py --random=3450808900320758527

```bash
2025-05-09T14:53:52.973000Z TestFramework (INFO): User supplied random seed 3450808900320758527
2025-05-09T14:53:52.973000Z TestFramework (INFO): PRNG seed is: 3450808900320758527
2025-05-09T14:53:52.974000Z TestFramework (INFO): Initializing test directory /var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_1dgb1kvc
2025-05-09T14:53:52.992000Z TestFramework (INFO): This test is time consuming, please be patient
2025-05-09T14:53:52.992000Z TestFramework (INFO): Splitting inputs so we can generate tx's
2025-05-09T14:53:53.560000Z TestFramework (INFO): Finished splitting
2025-05-09T14:53:54.423000Z TestFramework (INFO): Testing estimates with single transactions.
2025-05-09T14:53:54.423000Z TestFramework (INFO): Will output estimates for 1/2/3/6/15/25 blocks
2025-05-09T14:53:54.423000Z TestFramework (INFO): Creating transactions and mining them with a block size that can't keep up
2025-05-09T14:53:56.340000Z TestFramework (INFO): Creating transactions and mining them at a block size that is just big enough
2025-05-09T14:53:58.810000Z TestFramework (INFO): Creating transactions and mining them with a block size that can't keep up
2025-05-09T14:54:00.682000Z TestFramework (INFO): Creating transactions and mining them at a block size that is just big enough
2025-05-09T14:54:02.972000Z TestFramework (INFO): Final estimates after emptying mempools
2025-05-09T14:54:02.988000Z TestFramework (INFO): Test fee_estimates.dat is flushed periodically
2025-05-09T14:54:03.839000Z TestFramework (INFO): Test fee rate estimation after restarting node with high MempoolMinFee
2025-05-09T14:54:04.700000Z TestFramework (INFO): Test acceptstalefeeestimates option
2025-05-09T14:54:05.129000Z TestFramework (INFO): Test reading old fee_estimates.dat
2025-05-09T14:54:05.869000Z TestFramework (INFO): Restarting node with fresh estimation
2025-05-09T14:54:06.413000Z TestFramework (INFO): Testing estimates with RBF.
2025-05-09T14:54:08.411000Z TestFramework (INFO): Restarting node with fresh estimation
2025-05-09T14:54:09.016000Z TestFramework (INFO): Test estimatesmartfee modes
2025-05-09T14:54:27.128000Z TestFramework (INFO): Testing that fee estimation is disabled in blocksonly.
2025-05-09T14:54:27.607000Z TestFramework (INFO): Stopping nodes
2025-05-09T14:54:27.769000Z TestFramework (INFO): Cleaning up /var/folders/5t/04gq0pqj5yv4t8cxw51q0s2m0000gn/T/bitcoin_func_test_1dgb1kvc on exit
2025-05-09T14:54:27.769000Z TestFramework (INFO): Tests successful
```

--- a/test/functional/feature_fee_estimation.py
+++ b/test/functional/feature_fee_estimation.py
@@ -146,8 +146,8 @@ class EstimateFeeTest(BitcoinTestFramework):
         self.noban_tx_relay = True
         self.extra_args = [
             [],
-            ["-blockmaxweight=68000"],
-            ["-blockmaxweight=32000"],
+            ["-blockmaxweight=72000"],
+            ["-blockmaxweight=36000"],
         ]
 
     def setup_network(self):

--- a/test/functional/feature_fee_estimation.py
+++ b/test/functional/feature_fee_estimation.py
@@ -91,19 +91,20 @@ def check_smart_estimates(node, fees_seen):
     """Call estimatesmartfee and verify that the estimates meet certain invariants."""
 
     delta = 1.0e-6  # account for rounding error
-    last_feerate = float(max(fees_seen))
     all_smart_estimates = [node.estimatesmartfee(i) for i in range(1, 26)]
     mempoolMinFee = node.getmempoolinfo()["mempoolminfee"]
     minRelaytxFee = node.getmempoolinfo()["minrelaytxfee"]
+    feerate_ceiling = max(max(fees_seen), float(mempoolMinFee), float(minRelaytxFee))
+    last_feerate = feerate_ceiling
     for i, e in enumerate(all_smart_estimates):  # estimate is for i+1
         feerate = float(e["feerate"])
         assert_greater_than(feerate, 0)
         assert_greater_than_or_equal(feerate, float(mempoolMinFee))
         assert_greater_than_or_equal(feerate, float(minRelaytxFee))
 
-        if feerate + delta < min(fees_seen) or feerate - delta > max(fees_seen):
+        if feerate + delta < min(fees_seen) or feerate - delta > feerate_ceiling:
             raise AssertionError(
-                f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{max(fees_seen)})"
+                f"Estimated fee ({feerate}) out of range ({min(fees_seen)},{feerate_ceiling})"
             )
         if feerate - delta > last_feerate:
             raise AssertionError(
@@ -238,10 +239,10 @@ class EstimateFeeTest(BitcoinTestFramework):
         self.log.info("Final estimates after emptying mempools")
         check_estimates(self.nodes[1], self.fees_per_kb)
 
-    def test_feerate_mempoolminfee(self):
-        high_val = 3 * self.nodes[1].estimatesmartfee(1)["feerate"]
+    def test_estimates_with_highminrelaytxfee(self):
+        high_val = 3 * self.nodes[1].estimatesmartfee(2)["feerate"]
         self.restart_node(1, extra_args=[f"-minrelaytxfee={high_val}"])
-        check_estimates(self.nodes[1], self.fees_per_kb)
+        check_smart_estimates(self.nodes[1], self.fees_per_kb)
         self.restart_node(1)
 
     def sanity_check_rbf_estimates(self, utxos):
@@ -452,11 +453,11 @@ class EstimateFeeTest(BitcoinTestFramework):
         self.log.info("Test fee_estimates.dat is flushed periodically")
         self.test_estimate_dat_is_flushed_periodically()
 
-        # check that the effective feerate is greater than or equal to the mempoolminfee even for high mempoolminfee
+        # check that estimatesmartfee feerate is greater than or equal to maximum of mempoolminfee and minrelaytxfee
         self.log.info(
-            "Test fee rate estimation after restarting node with high MempoolMinFee"
+            "Test fee rate estimation after restarting node with high minrelaytxfee"
         )
-        self.test_feerate_mempoolminfee()
+        self.test_estimates_with_highminrelaytxfee()
 
         self.log.info("Test acceptstalefeeestimates option")
         self.test_acceptstalefeeestimates_option()

